src/engines/dropped-frames-watcher.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+import FakeEventTarget from '../event/fake-event-target';
+import FakeEvent from '../event/fake-event';
+import {CustomEventType} from '../event/event-type';
+import getLogger from '../utils/logger';
+
+const EXCEEDED_MAX_FRAME_DROP: string = 'exceededmaxframerop';
+
+class DroppedFramesWatcher extends FakeEventTarget {
+  _droppedFramesInterval: ?number = null;
+  _lastDroppedFrames: number = 0;
+  _lastDecodedFrames: number = 0;
+  _lastTime: number = 0;
+  _mediaSourceAdapter: IMediaSourceAdapter;
+  _config: Object = {};
+  _videoElement: HTMLVideoElement;
+  static _logger: any = getLogger('droppedFramesWatcher');
+
+  constructor(mediaSourceAdapter: IMediaSourceAdapter, config: Object, videoElement: HTMLVideoElement) {
+    super();
+    this._mediaSourceAdapter = mediaSourceAdapter;
+    this._config = config;
+    this._videoElement = videoElement;
+    this._droppedFramesInterval = setInterval(() => this._maybeAdjustBitrateQuality(), this._config.fpsDroppedFramesInterval);
+  }
+
+  _maybeAdjustBitrateQuality(): void {
+    if (typeof this._videoElement.getVideoPlaybackQuality === 'function') {
+      const videoPlaybackQuality = this._videoElement.getVideoPlaybackQuality();
+      this._checkFPS(videoPlaybackQuality.droppedVideoFrames, videoPlaybackQuality.totalVideoFrames);
+    } else {
+      // $FlowFixMe
+      this._checkFPS(this._videoElement.webkitDroppedFrameCount, this._videoElement.webkitDecodedFrameCount);
+    }
+  }
+
+  _checkFPS(droppedFrames: number, decodedFrames: number): void {
+    try {
+      const currentTime = performance.now();
+      if (decodedFrames) {
+        if (this._lastTime) {
+          const currentPeriod = currentTime - this._lastTime,
+            currentDropped = droppedFrames - this._lastDroppedFrames,
+            currentDecoded = decodedFrames - this._lastDecodedFrames,
+            droppedFPS = (1000 * currentDropped) / currentPeriod;
+          if (droppedFPS > 0) {
+            DroppedFramesWatcher._logger.debug('checkFPS : droppedFPS/decodedFPS:' + droppedFPS / ((1000 * currentDecoded) / currentPeriod));
+            if (currentDropped > this._config.fpsDroppedMonitoringThreshold * currentDecoded) {
+              const currentBandwidth = this._mediaSourceAdapter.getCurrentQuality();
```

Comment:
I do not see a reason to use this API like this as the data is already saved in player.
You can use an event and pass the player instance to listen to it on track change.
Also exposing this like this is very limited anyway - as if you have an API called `getCurrentQuality` you would expect to have the entire level object and be able to access level.bitrate on it.